modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -85,22 +91,33 @@ func (k Keeper) OnChanOpenTry(
 	}
 
 	// Register interchain account if it does not already exist
-	k.RegisterInterchainAccount(ctx, counterparty.PortId)
+	accAddr := strings.TrimPrefix(version, fmt.Sprintf("%s-", types.Version))
+	if err := k.RegisterInterchainAccount(ctx, accAddr, counterparty.PortId); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// OnChanOpenAck sets the active channel for the interchain account/owner pair
+// and stores the associated interchain account address in state keyed by it's corresponding port identifier
+//
+// Controller Chain
 func (k Keeper) OnChanOpenAck(
 	ctx sdk.Context,
 	portID,
 	channelID string,
 	counterpartyVersion string,
 ) error {
-	if counterpartyVersion != types.Version {
+	if !strings.Contains(counterpartyVersion, types.Version) {
 		return sdkerrors.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: %s, expected %s", counterpartyVersion, types.Version)
 	}
 
 	k.SetActiveChannel(ctx, portID, channelID)
 
+	accAddr := strings.TrimPrefix(counterpartyVersion, fmt.Sprintf("%s-", types.Version))
```

Comment:
I think we should call some kind of `ValidateVersionString(types.Version)` fn here that does a validation check for the expected format of a version string depending on the types.Version.

Reply:
Once we have validated that the format of the version string is what we expect, we can safely parse the accAddr.